packages/dds/merge-tree/src/test/revertibles.spec.ts

```diff
@@ -235,4 +239,107 @@ describe("MergeTree.Revertibles", () => {
 
         logger.validate();
     });
+
+    describe("Redo does not works as expected when a pair of markers and text is involved", ()=>{
+
+        generatePairwiseOptions({
+            revertMarkerInsert:[true, undefined],
+            ackMarkerInsert:[true, undefined], 
+            splitInsertTextRevertible:[true,undefined],
+            ackTextInsert:[true, undefined], 
+            splitRemoveRevertible:[true, undefined],
+            ackTextRemove:[true, undefined], 
+            ackUndo:[true, undefined]
+        }).forEach((options)=>{
+            it(JSON.stringify(options) , () => {
+                const clients = createClientsAtInitialState(
+                    { initialState: "", options: { mergeTreeUseNewLengthCalculations: true } },
+                    "A", "B");
+        
+                const logger = new TestClientLogger(clients.all);
+                let seq = 0;
+                const ops: ISequencedDocumentMessage[] = [];
+        
+                const clientB_Revertibles: MergeTreeDeltaRevertible[][] = [];
+                // the test logger uses these callbacks, so preserve it
+                const old = clients.B.mergeTreeDeltaCallback;
+                const clientBDriver = createRevertDriver(clients.B);
+                clientBDriver.submitOpCallback = (op) => ops.push(clients.B.makeOpMessage(op, ++seq));
+                clients.B.mergeTreeDeltaCallback = (op, delta) => {
+                    old?.(op, delta);
+                    if (op.sequencedMessage === undefined && clientB_Revertibles.length > 0) {
+                         appendToMergeTreeDeltaRevertibles(
+                            clientBDriver, 
+                            delta, 
+                            clientB_Revertibles[0]);
```

Comment:
I missed that you were grabbing the 0th entry of clientB_Revertibles on first read. took me a minute to realize the calls to `unshift` are to set up a new undo-redo transaction. I think the intent is a bit more obvious if you define a `const openNewUndoRedoTransaction = () => clientB_Revertibles.unshift([]);` and use that instead wherever you need.